autotest/autotest.cpp

```diff
@@ -2006,6 +2016,42 @@ bool tAutotest::step_dumpPackets(const YAML::Node& yamlStep,
 	return true;
 }
 
+bool tAutotest::step_bird(const YAML::Node& yamlStep,
+                          const std::string& path)
+{
+	std::string confPath;
+	for (const auto& yamlIter : yamlStep)
+	{
+		confPath = yamlIter.as<std::string>();
+	}
+
+	if (!confPath.empty())
+	{
+		const std::string execCommand = "bird -c " + path + confPath;
+
+		std::string resExec = exec(execCommand.c_str());
+
+		printf("%s", resExec.data());
+
+		return true;
+	}
+	return false;
+}
+
+bool tAutotest::step_birdc(const YAML::Node& yamlStep)
+{
+	std::string command = "birdc ";
+	for (const auto& yamlIter : yamlStep)
+	{
+		command += yamlIter.as<std::string>() + " ";
+	}
```

Comment:
Use `std::ostringstream` when building strings in a loop. Current approach is not efficient as you will gradually create more and more objects of increasing size.
```suggestion
    std::ostringstream command_stream;
    command_stream << "birdc ";

    for (const auto& yamlIter : yamlStep)
    {
        command_stream << yamlIter.as<std::string>() << ' ';
    }

    const std::string command = command_stream.str();
```